caikit/core/model_management/local_model_loader.py

```diff
@@ -0,0 +1,243 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The LocalModelLoader loads a model into local memory
```

Comment:
Clarification question:
1. What is the relation between "loader" and "backend" is that 1:1 or 1 to many? So like, Local loader will support multi-backends thus allowing a way to load models in TGIS etc?
2. I think the purpose of the "loader" is to go through the `config` returned by the "finder" and figure out which of the caikit modules can support it. After that the individual module can have backend implementations that the `backend` logic can manage. Is that accurate? If so, I think the logic related to `backend` probably belongs in the `base` class so that it can be shared across different "loaders"?

Reply:
Crap, yeah, this docstring is wrong since it may indeed load elsewhere. I'll fix it.

>If so, I think the logic related to backend probably belongs in the base class so that it can be shared across different "loaders"?

My reasoning for putting this in the `LocalModelLoader` rather than leaving it as a wrapper in `ModelManager` is that it doesn't really make sense to think about having an alternate `ModelLoader` that says "go load this thing over there" and then have _that_ be wrapped up with logic that does the multi-backend-implementation stuff. Basically, a given module will _either_ be loaded using some non-local `loader` OR with a non-local `backend`, but it's not clear what it would mean to try to do both together.

Reply:
To be fair, though, I think this really gets at the question of whether `module_backends` should even be supported anymore or if they should just be alternate `loaders`.